search/util.go

```diff
@@ -140,3 +140,9 @@ const KnnPreSearchDataKey = "_knn_pre_search_data_key"
 const PreSearchKey = "_presearch_key"
 
 type ScoreExplCorrectionCallbackFunc func(queryMatch *DocumentMatch, knnMatch *DocumentMatch) (float64, *Explanation)
+
+type SearchSearcherStartCallbackFn func(size uint64) error
+type SearchSearcherEndCallbackFn func(size uint64) error
+
+const SearchSearcherStartCallbackKey = "_search_searcher_start_callback_key"
+const SearchSearcherEndCallbackKey = "_search_searcher_end_callback_key"
```

Comment:
It seems we're applying all of these only for kNN at the moment - do you see it being used for anything else in the future that we keep the name as generic as this?

Reply:
yeah so this was like a future proof time of a thing - if we want to do the same for text search paths, we can use these keys and invoke at appropriate paths over there. this is assuming that in the future we decide that its in our interest to invoke the app_herder checks at those paths after a fair bit of system and volume testing.